frontend/src/components/user/SignUp.js

```diff
@@ -1,9 +1,109 @@
 import React from "react";
+import { Form, Input, Icon, Button } from "antd";
 
-const SignUp = () => (
-    <div>
-        <h1>Sign up below:</h1>
-    </div>
-);
+class SignUp extends React.Component {
+    constructor(props) {
+        super(props);
 
-export default SignUp;
+        this.state = {
+            first: "",
+            last: "",
+            email: "",
+            user: "",
+            password: ""
+        };
+    }
+
+    handleSubmit = e => {
+        e.preventDefault();
+        this.props.form.validateFields((err, values) => {
+            if (!err) {
+                console.log("Received values of form: ", values);
+                //do signup
+            }
+        });
+    };
+
+    render() {
+        const { getFieldDecorator } = this.props.form;
+
+        return (
+            <div style={{ margin: "auto", maxWidth: "400px" }}>
+                <h1>Sign up below:</h1>
+
+                <Form onSubmit={this.handleSubmit}>
+                    <Form.Item label='First Name'>
+                        {getFieldDecorator("first", {
```

Comment:
Maybe it is better if we store the result in a constant and then use it here for readability.

Reply:
What do you mean?

Reply:
```jsx
const fieldDecorator = getFieldDecorator(...);

return (
  ...
  <Form.Item>
    {fieldDecorator}
  </Form.Item>
  ...
);
```